### PR TITLE
PLT-7646 Fix progress % values for properties in the Timeline-View grid

### PR DESCRIPTION
**Pull request by amnambiar**
https://input-output.atlassian.net/browse/PLT-7646

BE PR https://github.com/input-output-hk/dapps-certification/pull/90

### Summary by CodeRabbit

- New Feature: Introduced `CircularProgressBar` and `ProgressCard` components for better progress visualization.
- New Feature: Added `RepoAccessStatus` component to display repository access status.
- New Feature: Implemented `StatusIcon` component for displaying various status icons.
- Refactor: Simplified routing structure in `App.tsx` and removed unused imports and components.
- Refactor: Updated `Button` component to support passing an icon element.
- Refactor: Enhanced `TextArea` component with onBlur validation and floating label functionality.
- Refactor: Improved `TableComponent` to support collapsible rows and show/hide column functionality.
- Refactor: Modified `ColViz` component for better visual presentation and interaction.
- Refactor: Updated `useLocalStorage` hook for better error handling and value type flexibility.
- Bug Fix: Fixed image paths in `repoAccessStatus.config.ts`.
- Test: Commented out test cases for `LogsView` component.
- Chore: Updated CSS styles across multiple components for improved UI.

### CODE REVIEW

File: src/pages/certification/certification-result/CertificationResult.tsx [hunk 1]
```diff
@@ -52,6 +55,16 @@ const CertificationResult = () => {
...
+          const resultTaskKeys = Object.keys(resultJson)
+          const certificationTaskKeys = taskKeys();
+          let resultTaskKeysLength = 0;
+          resultTaskKeys.forEach((key: any) => {
+            if (certificationTaskKeys.indexOf(key) !== -1 && resultJson[key]) {
+              currentPropertyBasedTestProgress.current += isTaskSuccess(resultJson[key], key) ? 100 : 0
+              resultTaskKeysLength++;
+            }
+          })
+          propertyBasedTestProgressTotal.current = resultTaskKeysLength * 100
```
Comment (RSoulatIOHK) [inline src/pages/certification/certification-result/CertificationResult.tsx]: That only "works" when the properties are a success.

**Comment by amnambiar** [comment]
@RSoulatIOHK Resolved the calculations within the progress% in TimelineView ; and added the Property Based Testing progress card in both Timeline View and Full Report View when run is finished.


Calculations are updated here - https://input-output.atlassian.net/browse/PLT-7646?focusedCommentId=217176

File: src/pages/certification/Certification.helper.tsx [hunk 2]
```diff
@@ -53,6 +100,18 @@ export const isAnyTaskFailure = (result: any) => {
...
+    failed = result.filter((item: any) =>  item[1].tag === 'Failure')?.length ? true : false
+  } else {
+    failed = result.tag === "Failure"
+  }
+  return !failed
+}
+
 export const processTablesDataForChart = (resultObj: any, tableAttr: string) => {
   let totalCount = 0
   try {
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/Certification.helper.tsx:117]: not intended.ignoring

File: src/pages/certification/Certification.helper.tsx [hunk 3]
```diff
@@ -93,4 +152,74 @@ export const getCertificationTaskName = (key: string) => {
...
+          }
+        })
+      } else {
+        item.expected = DEFAULT_TESTS_COUNT
+        item.completed = keyResult.numTests || 0
+      }
+    }
+  }
+  return item
 }
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/pages/certification/Certification.helper.tsx:225]: ignoring for now

File: src/components/LogsView/LogsView.tsx [hunk 4]
```diff
@@ -4,86 +4,102 @@ import React, { FC, useState, useEffect, useRef } from "react";
...
+                // do nothing
+                if (typeof item.Text == "string" && item.Text.length) {
+                  logData = item.Text;
+                }
+              }
+              logData = !logData.length ? item.Text : logData;
+              return logData.length ? (
+                <LogsViewEntry key={index} time={item.Time} log={logData} />
+              ) : null;
+            })}
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/components/LogsView/LogsView.tsx:96]: ignoring for now

**Review by RSoulatIOHK — APPROVED** [review]

File: src/components/LogsView/LogsView.tsx [hunk 5]
```diff
@@ -4,86 +4,102 @@ import React, { FC, useState, useEffect, useRef } from "react";
...
+  useEffect(() => {
+    if (endPolling && latestTestingProgress) {
+      lastLog.current = logs[logs.length - 2]
+      if (lastLog.current?.Source.indexOf('run-certify') !== -1) {
+        // is 2nd last entry in logs when Source has run-certify
+        latestTestingProgress(JSON.parse(lastLog.current.Text));
+      }
+    }
+    bottomRef.current?.scrollIntoView({ behavior: "smooth" }); // scroll to bottom
+  }, [logs, endPolling, latestTestingProgress]);
```
[the remark this answers is not in the document]
Comment (amnambiar) [reply src/components/LogsView/LogsView.tsx:48]: ignoring for now

File: src/pages/certification/certification-result/CertificationResult.tsx (hunk 1 above)
Comment (amnambiar) [reply src/pages/certification/certification-result/CertificationResult.tsx]: That's an outdated commit.

**Pull request merged**